nexus/types/src/external_api/params.rs

```diff
@@ -474,7 +474,7 @@ impl UserData {
     where
         D: Deserializer<'de>,
     {
-        match base64::decode(<&str>::deserialize(deserializer)?) {
+        match base64::decode(<String>::deserialize(deserializer)?) {
```

Comment:
tl;dr: I think you can ignore this, but here are the gory details.

Without this change, I saw lots of failures because we could not deserialize `params::InstanceCreate` from a `serde_json::Value`.  The error looked like this:

```
invalid type: string \"\", expected a borrowed string
```

This was reproducible with this test in params.rs:

```rust
#[test]
fn test_dap2() {
    let input = serde_json::json! {
        {   
          "name": "i1",
          "description": "asdf",
          "ncpus": 1,
          "memory": 1073741824,
          "hostname": "asdf",
          "user_data": "", 
          "network_interfaces": {
            "type": "default"
          },  
          "external_ips": [], 
          "disks": []
        }   
    };  
    let busted_as_string = serde_json::to_string_pretty(&input).unwrap();
    println!("busted input:\n{}", busted_as_string);
    println!("\nresult of parsing the serde_json::Value:");
    println!("{:?}", serde_json::from_value::<InstanceCreate>(input));
    println!("\nresult of parsing the same value, serialized to a string:");
    println!("{:?}", serde_json::from_str::<InstanceCreate>(&busted_as_string));
}
```

which produced:

```
busted input:
{
  "description": "asdf",
  "disks": [],
  "external_ips": [],
  "hostname": "asdf",
  "memory": 1073741824,
  "name": "i1",
  "ncpus": 1,
  "network_interfaces": {
    "type": "default"
  },
  "user_data": ""
}

result of parsing the serde_json::Value:
Err(Error("invalid type: string \"\", expected a borrowed string", line: 0, column: 0))

result of parsing the same value, serialized to a string:
Ok(InstanceCreate { identity: IdentityMetadataCreateParams { name: Name("i1"), description: "asdf" }, ncpus: InstanceCpuCount(1), memory: ByteCount(1073741824), hostname: "asdf", user_data: [], network_interfaces: Default, external_ips: [], disks: [] })
```

I'm not exactly sure what a `Deserialize` impl is _supposed_ to do here.  The [serde docs on Deserializer lifetimes](https://serde.rs/lifetimes.html) say:

> Not all deserializers support handing out borrowed data. For example when deserializing from an IO stream no data can be borrowed.

Given that, it seems safer to avoid deserializing borrowed values.

Why did I hit this now?  I think it's because with this change, the instance create saga stores `params::InstanceCreate` and deserializes them during the saga.  That causes them to go through a `serde_json::Value` where I suspect they didn't before.  However, the saga params _do_ go through a `serde_json::Value` at least during recovery so I suspect that saga recovery for the instance create saga was already broken.  I haven't tested that.